libraries/cli11/include/cli11/CLI11.hpp

```diff
@@ -9420,4 +9420,134 @@ inline std::string Formatter::make_option_usage(const Option *opt) const {
+         }
+         std::vector<const App*> subcommands_group = app->get_subcommands([&group](const App* sub_app) {
+            return detail::to_lower(sub_app->get_group()) == detail::to_lower(group);
+         });
+         for(const App* new_com: subcommands_group) {
+            if(new_com->get_name().empty())
+               continue;
+
+            std::string tree_symbol = (subcommands_group.back() == new_com ? tree_angle : tree_fork);
+            std::string line_symbol = (subcommands_group.back() == new_com ? "" : tree_line);
+            std::string subc_symbol = "";
+
+            const App* parent = app->get_parent();
+            if(parent != nullptr) {
+               std::vector<const App*> sc_group = parent->get_subcommands([&group](const App* sub_app) {
+                  return detail::to_lower(sub_app->get_group()) == detail::to_lower(group);
+               });
+               if(sc_group.back() != app) {
+                  subc_symbol = tree_line;
+               }
+            }
+
+            switch(mode) {
+               case AppFormatMode::All:
+                  out << tree_symbol << new_com->help(new_com->get_name(), AppFormatMode::Sub);
+                  out << "\n";
+                  break;
+               case AppFormatMode::AllCompact:
+
+                  out << tree_symbol << new_com->help(new_com->get_name(), AppFormatMode::SubCompact);
+                  out << line_symbol;
+                  out << "\n";
+                  break;
+               case AppFormatMode::Normal:
+               case AppFormatMode::Sub:
+                  out << make_subcommand(new_com);
+                  break;
+               case AppFormatMode::SubCompact:
+                  out << tree_symbol << make_expanded(new_com, mode);
+                  break;
+               default:
+                  throw HorribleError("Internal error: unknown help type requested");
+            }
+         }
+      }
+
+      return out.str();
+   }
+
+   /// This prints out a subcommand in help-all
+   virtual std::string make_expanded(const App* sub, AppFormatMode mode = AppFormatMode::Sub) const {
+      std::stringstream out;
+      std::string tmp;
+      std::string subc_symbol = " ";
+      if(mode == AppFormatMode::SubCompact) {
+         detail::format_help(out, sub->get_display_name(true), sub->get_description(), column_width_);
+         out << make_subcommands(sub, mode);
+      } else {
+         out << sub->get_display_name(true) << "\n";
+         out << make_description(sub);
+         if(sub->get_name().empty() && !sub->get_aliases().empty()) {
+            detail::format_aliases(out, sub->get_aliases(), column_width_ + 2);
+         }
+         out << make_positionals(sub);
+         out << make_groups(sub, mode);
+         out << make_subcommands(sub, mode);
+      }
+
+      // Drop blank spaces
+      tmp = detail::find_and_replace(out.str(), "\n\n", "\n");
+      tmp = tmp.substr(0, tmp.size() - 1);// Remove the final '\n'
```

Comment:
Are we sure the end of the string is always '\n'?

Reply:
Yes in this context. Thank you!